### DISCUSSION TITLE
How to bind model property to DataGridTemplate column (Combobox)

### DISCUSSION THREAD

**Discussion by MichailBarsky** [AvaloniaUI/Avalonia#13032] [Q&A] [answered]
I'm new to Avalonia, but I have a lot of experience developing for .Net. I just can’t figure out how to bind one of the properties of a model to a DataGridTemplateColumn (Combobox). I understand how to bind a property to DataGridTextColumn (**{Binding PropertyName}**), but with ComboBox it’s completely unclear

As an example, let's say we have two models: “People” and “Sexes”. The "People" model references the "Sexes" model, for example through the foreign key Sex_id. So, how to implement a combobox column for the Sex_id property in the DataGrid for the “People” model

**Comment by timunie** [comment, marked as answer]
```xml
<DataGrid x:Name="root" ItemsSoruce="{Binding People}">
  <DataGrid.Columns>
    <DataGridTemplateColumn>
       <ComboBox ItemsSource="{Binding #root.((vm:MainViewModel)DataContext).Sexes}" SelectedItem="..." />
[...]
```

**Reply by MichailBarsky** [reply]
Thank you very much, but I think this is not all I need. This example specifies the data source for the combobox itself, this is clear to me, but how to bind the selected value itself to the corresponding property. (In our example, this is the Sex_id field of the People model). I understand that somehow I need to use the DisplayMemberBinding and SelectedValueBinding properties, but I can’t seem to bind SelectedValue to the property

**Reply by MichailBarsky** [reply]
Something like this (this is .Net Framework + WinForms)
![relations](https://github.com/AvaloniaUI/Avalonia/assets/43564200/a7206431-9c99-43bc-b4bd-3c55575475e2)
![table](https://github.com/AvaloniaUI/Avalonia/assets/43564200/37e0c49f-8aef-40b3-9f37-23fbae54f420)

**Comment by MichailBarsky** [comment]
I figured out how to do it. Thanks for the tip !
